File: services/ai-storage-service/src/models/Conversation.js
const mongoose = require('mongoose');

const messageSchema = new mongoose.Schema(
  {
    role: { type: String, enum: ['user', 'assistant', 'system'], required: true },
    content: { type: String, required: true },
  },
  { _id: false, timestamps: true }
);

const conversationSchema = new mongoose.Schema(
  {
    ownerId: { type: String, required: true, index: true },
    title: { type: String, default: 'New conversation' },
    provider: { type: String },
    model: { type: String },
    messages: { type: [messageSchema], default: [] },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Conversation', conversationSchema);

File: services/ai-storage-service-python/models/conversation.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from bson import ObjectId


class Message(BaseModel):
    role: str = Field(..., pattern="^(user|assistant|system)$")
    content: str
    timestamp: Optional[datetime] = None


class Conversation(BaseModel):
    id: Optional[str] = Field(None, alias="_id")
    owner_id: str
    title: str = "New conversation"
    provider: Optional[str] = None
    model: Optional[str] = None
    messages: List[Message] = []
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    
    class Config:
        populate_by_name = True
        json_encoders = {ObjectId: str}


class ConversationCreate(BaseModel):
    owner_id: str
    title: str = "New conversation"
    provider: Optional[str] = None
    model: Optional[str] = None
    messages: List[Message] = []


class ConversationUpdate(BaseModel):
    title: Optional[str] = None
    messages: Optional[List[Message]] = None
